pkg/resources/data.go

```diff
@@ -686,22 +686,24 @@ func GetCSIMigrationFeatureGates(cluster *kubermaticv1.Cluster, version *semverl
 		// The following feature gates are always enabled when the
 		// 'externalCloudProvider' feature is activated.
 		if cluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] {
-			featureFlags = append(featureFlags, "CSIMigration=true", "ExpandCSIVolumes=true")
+			featureFlags = append(featureFlags, "CSIMigration=true")
 		}
 
 		// This flag is GA since 1.24 and enabled by default; it seems to be gone
 		// from Kubernetes 1.26, so we don't need to set it anymore.
-		if cluster.Spec.Cloud.Openstack != nil && lt25.Check(version) {
-			featureFlags = append(featureFlags, "CSIMigrationOpenStack=true")
-		}
-		if cluster.Spec.Cloud.VSphere != nil {
-			featureFlags = append(featureFlags, "CSIMigrationvSphere=true")
-		}
-		if cluster.Spec.Cloud.AWS != nil {
-			featureFlags = append(featureFlags, "CSIMigrationAWS=true")
-		}
-		if cluster.Spec.Cloud.Azure != nil {
-			featureFlags = append(featureFlags, "CSIMigrationAzureDisk=true", "CSIMigrationAzureFile=true")
+		if lt25.Check(version) {
+			if cluster.Spec.Cloud.Openstack != nil {
+				featureFlags = append(featureFlags, "CSIMigrationOpenStack=true")
+			}
+			if cluster.Spec.Cloud.VSphere != nil {
+				featureFlags = append(featureFlags, "CSIMigrationvSphere=true")
+			}
+			if cluster.Spec.Cloud.AWS != nil {
+				featureFlags = append(featureFlags, "CSIMigrationAWS=true")
+			}
```

Comment:
CSIMigrationAWS is enabled by default starting with Kubernetes 1.23, so we can completely remove this `if`: https://github.com/kubernetes/kubernetes/blob/1635c380b26a1d8cc25d36e9feace9797f4bae3c/pkg/features/kube_features.go#L887